### ISSUE TITLE
[Error 403] On every Cloudflare site via Linkem's IP

### ISSUE THREAD

**Issue by FedericoHeichou** [inorichi/tachiyomi#1890]
As by title, I receive Error 403 if I use Wi-Fi (my ISP is Linkem), over "data connection" it works as normally.
The biggest part of catalogs returns me a 403 error, I think that happens when they use Cloudflare (2 of them uses Cloudflare for sure), because today, while I was navigating on the Internet from PC, I received 3 captcha request from Cloudflare.
So I thought that Cloudflare did something.
Casually today I saw Tachiyomi 0.8.1 was released, and in log change there is:

> Fixed latest Cloudflare changes.

Maybe it's more a problem of my ISP's IP addresses pool, but I preferred to open an issue anyway, maybe could be done more improvements to bypass this problem, maybe not, who knows, in the doubt I report this.

I opened it here instead of extensions repo because of that "Fixed latest Cloudflare changes" in update log.

More information about: websites like nhentai.net ask me for captcha on the PC (they set an high protection on Cloudflare), while websites like lupiteam.net or jaiminisbox.com don't ask anything on PC (they set a low protection on Cloudflare), but both cases return 403 on the app.
It's for sure a problem about untrusted IP, but I think there are others elements that don't helps with the connection. I don't know which they're, but I think they could be connection parameters like HTTP_REFERER, user-agent, screen size, etc... Those sent from the app are untrusted for Cloudflare compared to those of Desktop Chromium.

Basically imho "latest Cloudflare changes" isn't 100% fixed.

I have everything updated (latest version of each extension and tachiyomi 0.8.1).

**Comment by CarlosEsco** [comment]
Sounds like 100% a your ip issue.  Have you tried on a different network.

**Comment by FedericoHeichou** [comment]
Yes, as I said on different network it works.
It's an IP issue, but on others agents (like Chromium or any android browser) I don't get problems on those website that have enabled "low level of protection" on their Cloudflare panel.

So, it's 100% IP issue on those websites with high level of protection like nhentai.net.
It's 50% IP and 50% app on those websites that have a low level of protection.
Those websites with low level of protection should works on Tachiyomi, because on browser they do.

I think Cloudflare assigns a score to each connection to see if ask a captcha or not, and a bad IP + a strange client like Tachiyomi give me a bad score that is too low to see those websites with low level of protection too.

**Comment by CarlosEsco** [comment]
Why not build an extension and verify your theory then?  Extensions can control their own headers.

**Comment by DitFranXX** [comment]
CF issue in 0.8.1 is related to this, not captcha.
![image](https://user-images.githubusercontent.com/45893338/54468120-02104800-47cd-11e9-9fb3-bab90b9932cb.png)

Captcha is different issue.

**Comment by FedericoHeichou** [comment]
I think I have to try by myself as Carlos suggested, because you can't test.
For the moment I have no time to do this, so if the problem will persist I will do in the future and I will reply to this issue with the solution.

Ok, so this is a new "Cloudflare captcha issue".

I will let you know

**Comment by arkon** [comment]
This is pretty old and it sounds like an ISP issue.
